### PR TITLE
Ideas for issues, PRs and others

### PR DESCRIPTION
**Pull request by luizcmarin**
I created some ideas for issues/PRs and others. Please evaluate.

See  too https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates

### CODE REVIEW

File: UPGRADE.md
Comment (vjik) [inline UPGRADE.md:1]: This file not need for new packages.

File: .github/ISSUE_TEMPLATE.yml
Comment (vjik) [inline .github/ISSUE_TEMPLATE.yml:1]: Seems, it is duplicate #111

File: .github/PULL_REQUEST_TEMPLATE.yml
Comment (vjik) [inline .github/PULL_REQUEST_TEMPLATE.yml:1]: I think is to much for PR currently.

**Review by vjik — CHANGES_REQUESTED** [review]

File: UPGRADE.md
Comment (luizcmarin) [reply UPGRADE.md:1]: I imagine, I just imagine, that this package is a template for all new packages. So, I just thought about having as complete a package as possible, including the 'upgrade.md' file as a template. If the developer doesn't need it, they delete it.

File: .github/PULL_REQUEST_TEMPLATE.yml
Comment (luizcmarin) [reply .github/PULL_REQUEST_TEMPLATE.yml:1]: I agree with you. It was too good!

File: UPGRADE.md
Comment (vjik) [reply UPGRADE.md:1]: This file never needs in new package.

Comment (luizcmarin) [reply UPGRADE.md:1]: I will remove

File: .github/ISSUE_TEMPLATE.yml
Comment (luizcmarin) [reply .github/ISSUE_TEMPLATE.yml:1]: yes, it is duplicated. still learning how to deal with Github. If I close this other PR, will I lose the comments that are there?

File: .github/PULL_REQUEST_TEMPLATE.yml
Comment (vjik) [reply .github/PULL_REQUEST_TEMPLATE.yml:1]: Can you show screenshot, how it looks?

File: .github/ISSUE_TEMPLATE.yml
Comment (vjik) [inline .github/ISSUE_TEMPLATE.yml:1]: Can you show screenshot, how it looks?

**Comment by luizcmarin** [comment]
I kept HTML in this file because it is not frequently modified and for its visual impact.
image

Best I got. I didn't find a decent preview.
image

File: .github/PULL_REQUEST_TEMPLATE.yml
Comment (luizcmarin) [reply .github/PULL_REQUEST_TEMPLATE.yml:1]: the best I got in the images below

**Comment by vjik** [comment]
I think currently simple template for PR and issues is OK.

I don't like HTML in md-files, it is too much for me.

**Comment by luizcmarin** [comment]
And these files?   How are they?
- PULL_REQUEST_TEMPLATE.md
- ISSUE_TEMPLATE.md

File: .github/ISSUE_TEMPLATE.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  > ***WARNING!*** DO NOT use the issue tracker or discuss it in the public forum as it will cause more damage than help.
+description: Create a bug report
+labels: ["defect"]
+body:
+  - type: input
+    id: yiiversion
+    attributes:
+      label: Yii Version
+      description: Which version of Yii are you using?
+      placeholder: "Enter the version you are using of this package"
```
Comment (samdark) [inline .github/ISSUE_TEMPLATE.yml:13]:
```suggestion
      placeholder: "Enter the version of this package you are using"
```

File: .github/ISSUE_TEMPLATE.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      description: What version of operating system are you running?
+      placeholder: "Linux (Default)"
+      multiple: false
+      options:
+        - Microsoft Windows
+        - Apple macOS
+        - Linux (Default)
+        - Android
+        - iOS
+        - Chrome OS
```
Comment (samdark) [inline .github/ISSUE_TEMPLATE.yml:37]:
```suggestion
PHP doesn't really run on these. We should remove it.
```

File: .github/PULL_REQUEST_TEMPLATE.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Pull Request
+literal_block: |
+  > If you have encountered a security-related issue with Yii, please use the [security issue form](https://www.yiiframework.com/security) to report any security issues you have encountered to us.
+  > ***WARNING!*** DO NOT use the issue tracker or discuss it in the public forum as it will cause more damage than help.
```
Comment (samdark) [inline .github/PULL_REQUEST_TEMPLATE.yml:4]: This isn't relevant for pull requests.

File: .github/PULL_REQUEST_TEMPLATE.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+description: Create a pull request to propose and collaborate on changes to this repository.
+labels: ["suggestion"]
+body:
+  - type: textarea
+    id: expected
+    attributes:
+      label: Expected result
```
Comment (samdark) [inline .github/PULL_REQUEST_TEMPLATE.yml:11]: Not needed for a pull request.

File: .github/ISSUE_TEMPLATE.yml (hunk 2 above)
Comment (luizcmarin) [reply .github/ISSUE_TEMPLATE.yml:37]: add "PHP doesn't really run on these. We should remove it." ?

Comment (luizcmarin) [reply .github/ISSUE_TEMPLATE.yml]: done. review

File: .github/PULL_REQUEST_TEMPLATE.yml (hunk 3 above)
Comment (luizcmarin) [reply .github/PULL_REQUEST_TEMPLATE.yml:4]: done

File: .github/PULL_REQUEST_TEMPLATE.yml (hunk 4 above)
Comment (luizcmarin) [reply .github/PULL_REQUEST_TEMPLATE.yml:11]: done

**Comment by vjik** [comment]
> And these files? How are they?
> 
> * PULL_REQUEST_TEMPLATE.md
> * ISSUE_TEMPLATE.md

Yes. Keep these files as is. I think currently enough simple templates.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,5 +1 @@
-| Q             | A
-| ------------- | ---
-| Is bugfix?    | ✔️/❌
-| New feature?  | ✔️/❌
-| Breaks BC?    | ✔️/❌
+### Additional info
```
[the remark this answers is not in the document]
Comment (luizcmarin) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Adjusted.

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -1,16 +1,7 @@
...
-
-### What is the expected result?
-
-### What do you get instead?
-
+| Q             | A      |
+| ------------- | ------ |
+| Is bugfix?    | ✔️/❌ |
+| New feature?  | ✔️/❌ |
+| Breaks BC?    | ✔️/❌ |
```
Comment (samdark) [inline .github/ISSUE_TEMPLATE.md:5]: This one is for pull requests. It doesn't make sense for issues.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,5 +1,16 @@
...
+### What do you get instead?
+
+
+### Additional info
+
+| Q                | A     |
+| ---------------- | ----- |
+| Version          | 1.0.? |
+| PHP version      |       |
+| Operating system |       |
```
Comment (samdark) [inline .github/PULL_REQUEST_TEMPLATE.md:16]: This is for issues. Doesn't make sense for pull requests.

Comment (luizcmarin) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: I reversed... 🤷🏻‍♂️🤦🏻‍♂️

File: .github/ISSUE_TEMPLATE.md (hunk 6 above)
Comment (luizcmarin) [reply .github/ISSUE_TEMPLATE.md:5]: I reversed... 🤷🏻‍♂️🤦🏻‍♂️

**Review by samdark — APPROVED** [review]
Looks OK to me.

File: .github/ISSUE_TEMPLATE.yml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+literal_block: |
+  > If you have encountered a security-related issue with Yii, please use the [security issue form](https://www.yiiframework.com/security) to report any security issues you have encountered to us.
+  > ***WARNING!*** DO NOT use the issue tracker or discuss it in the public forum as it will cause more damage than help.
+description: Create a bug report
+labels: ["defect"]
+body:
+  - type: input
+    id: yiiversion
+    attributes:
+      label: Yii Version
```
Comment (vjik) [inline .github/ISSUE_TEMPLATE.yml:11]: This is package template. This field not need.

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -21,3 +23,20 @@ If you don't have any particular package in mind to start with:
...
+
+If you have encountered a security-related issue with Yii, please use the [security issue form](https://www.yiiframework.com/security) to report any security issues you have encountered to us.
+
+***WARNING!*** DO NOT use the issue tracker or discuss it in the public forum as it will cause more damage than help.
+
+# Additional Resources
+
+- [Yii coding standards](https://github.com/yiisoft/docs/blob/master/010-code-style.md)
+- [Forum](https://www.yiiframework.com/forum/)
+- [Yii3 Existing issues](https://forum.yiiframework.com/c/yii-3-0/63)
```
Comment (vjik) [inline .github/CONTRIBUTING.md:37]: This is forum, not issues.

File: .github/PULL_REQUEST_TEMPLATE.yml
Comment (vjik) [inline .github/PULL_REQUEST_TEMPLATE.yml:1]: Can you create repo with new PR and Issue templates?

**Review by vjik — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 9 above)
Comment (luizcmarin) [reply .github/CONTRIBUTING.md:37]: remove "[Forum](https://www.yiiframework.com/forum/)" ?
add something else?

File: .github/ISSUE_TEMPLATE.yml (hunk 8 above)
Comment (luizcmarin) [reply .github/ISSUE_TEMPLATE.yml:11]: But add/maintain this in other packages?

File: .github/CONTRIBUTING.md (hunk 9 above)
Comment (vjik) [reply .github/CONTRIBUTING.md:37]: Seems, "Yii3 Existing issues" is a invalid label.

File: .github/ISSUE_TEMPLATE.yml (hunk 8 above)
Comment (vjik) [reply .github/ISSUE_TEMPLATE.yml:11]: If we use separate fields, then this should be conrecte package version. For example, "Yii DB version".

**Pull request closed without merging**